code/contracts/CommitManager.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.4;
+
+/// we will slowly move commit-reveal related stuff from ONEWallet to here
+library CommitManager {
+    uint32 constant REVEAL_MAX_DELAY = 60;
+
+    struct Commit {
+        bytes32 paramsHash;
+        bytes32 verificationHash;
+        uint32 timestamp;
+        bool completed;
+    }
+
+    struct CommitState {
+        mapping(bytes32 => Commit[]) commitLocker;
+        bytes32[] commits;  // self-clean on commit (auto delete commits that are beyond REVEAL_MAX_DELAY), so it's bounded by the number of commits an attacker can spam within REVEAL_MAX_DELAY time in the worst case, which is not too bad.
+    }
+
+
+    function getNumCommits(mapping(bytes32 => Commit[]) storage commitLocker, bytes32[] storage commits) view internal returns (uint32){
+        uint32 numCommits = 0;
+        for (uint32 i = 0; i < commits.length; i++) {
+            numCommits += uint32(commitLocker[commits[i]].length);
+        }
+        return numCommits;
+    }
+
+    function getCommitHashes(CommitState storage cs, uint32 numCommits) internal view returns (bytes32[] memory){
```

Comment:
functions intended to be used only inside the library should be prepended with `_`

Reply:
to be fixed in next version